### PR TITLE
Add logs for catalog

### PR DESCRIPTION
**Pull request by linyguo**
- Added context logs for catalog provider
- Updated logs in configs and secrets manager
- Went through parser.go and add logs in the recursive functions. Errors in internal functions will be returned and be handled and logged in `evalProperties` and `enumerateProperties`.

### CODE REVIEW

File: api/pkg/apis/v1alpha1/providers/config/catalog/catalogprovider.go [hunk 1]
```diff
@@ -95,6 +95,7 @@ func (m *CatalogConfigProvider) unwindOverrides(ctx context.Context, override st
 	override = utils.ConvertReferenceToObjectName(override)
 	catalog, err := m.ApiClient.GetCatalog(ctx, override, namespace, m.Config.User, m.Config.Password)
 	if err != nil {
+		clog.ErrorCtx(ctx, " P (Catalog): Unwind overrides error:", err)
```
Comment (LeiQL) [inline api/pkg/apis/v1alpha1/providers/config/catalog/catalogprovider.go:98]: Two spaces before `P`. Check `helm.go` or other providers logging.

Comment (linyguo) [reply api/pkg/apis/v1alpha1/providers/config/catalog/catalogprovider.go]: updated.

File: coa/pkg/apis/v1alpha2/providers/config/config.go [hunk 2]
```diff
@@ -23,6 +23,6 @@ type IConfigProvider interface {
 }
 
 type IExtConfigProvider interface {
-	Get(object string, field string, overrides []string, localContext interface{}) (interface{}, error)
-	GetObject(object string, overrides []string, localContext interface{}) (map[string]interface{}, error)
+	Get(ctx context.Context, object string, field string, overrides []string, localContext interface{}) (interface{}, error)
```
Comment (msftcoderdjw) [inline coa/pkg/apis/v1alpha2/providers/config/config.go:26]: We only have configs-manager as IExtConfigProvider's implementation? right?

**Review by msftcoderdjw — DISMISSED** [review]

File: coa/pkg/apis/v1alpha2/providers/config/config.go (hunk 2 above)
Comment (linyguo) [reply coa/pkg/apis/v1alpha2/providers/config/config.go:26]: yes, all three interfaces has updated context.

File: api/pkg/apis/v1alpha1/managers/configs/configs-manager.go [hunk 3]
```diff
@@ -30,9 +30,14 @@ type ConfigsManager struct {
...
-func (s *ConfigsManager) Init(context *contexts.VendorContext, cfg managers.ManagerConfig, providers map[string]providers.IProvider) error {
-	log.Debug(" M (Config): Init")
-	err := s.Manager.Init(context, cfg, providers)
+func (s *ConfigsManager) Init(vendorCtx *contexts.VendorContext, cfg managers.ManagerConfig, providers map[string]providers.IProvider) error {
+	ctx := context.TODO()
+	if vendorCtx != nil && vendorCtx.EvaluationContext != nil && vendorCtx.EvaluationContext.Context != nil {
+		ctx = vendorCtx.EvaluationContext.Context
+	}
+
+	log.DebugCtx(ctx, " M (Config): Init")
```
Comment (iwangjintian) [inline api/pkg/apis/v1alpha1/managers/configs/configs-manager.go:39]: we don't use CTX log in init functions

File: api/pkg/apis/v1alpha1/managers/secrets/secrets-manager.go [hunk 4]
```diff
@@ -30,9 +30,14 @@ type SecretsManager struct {
 	Precedence      []string
 }
 
-func (s *SecretsManager) Init(context *contexts.VendorContext, cfg managers.ManagerConfig, providers map[string]providers.IProvider) error {
-	log.Debug(" M (secret): Init")
-	err := s.Manager.Init(context, cfg, providers)
+func (s *SecretsManager) Init(vendorCtx *contexts.VendorContext, cfg managers.ManagerConfig, providers map[string]providers.IProvider) error {
+	ctx := context.TODO()
```
Comment (iwangjintian) [inline api/pkg/apis/v1alpha1/managers/secrets/secrets-manager.go:34]: Same here.

File: api/pkg/apis/v1alpha1/providers/config/catalog/catalogprovider.go [hunk 5]
```diff
@@ -188,6 +195,11 @@ func (m *CatalogConfigProvider) ReadObject(ctx context.Context, object string, l
 }
 
 func (m *CatalogConfigProvider) traceValue(v interface{}, localcontext interface{}, dependencyList map[string]map[string]bool) (interface{}, error) {
+	logCtx := context.TODO()
```
Comment (iwangjintian) [inline api/pkg/apis/v1alpha1/providers/config/catalog/catalogprovider.go:198]: Why initialize a todo() here? I saw you give the ltx.Context value right after that.

File: api/pkg/apis/v1alpha1/utils/parser.go [hunk 6]
```diff
@@ -781,7 +781,7 @@ func (n *FunctionNode) Eval(context utils.EvaluationContext) (interface{}, error
 			if err != nil {
 				return nil, err
 			}
-			return context.SecretProvider.Get(FormatAsString(obj), FormatAsString(field), context)
+			return context.SecretProvider.Get(context.Context, FormatAsString(obj), FormatAsString(field), context)
```
Comment (iwangjintian) [inline api/pkg/apis/v1alpha1/utils/parser.go:784]: where does this context.Context come from? We should get it from the evaluation context?

File: api/pkg/apis/v1alpha1/utils/parser.go [hunk 7]
```diff
@@ -1456,12 +1456,14 @@ func enumerateProperties(js interface{}, context utils.EvaluationContext) (inter
 				parser := NewParser(strVal)
 				val, err := parser.Eval(context)
 				if err != nil {
+					log.ErrorfCtx(context.Context, " (Parser): Enumerate properties failed: %v", err)
```
Comment (iwangjintian) [inline api/pkg/apis/v1alpha1/utils/parser.go:1459]: I believe parser has lots of error return but not logged. Do we choose these three places to expose the error?

File: api/pkg/apis/v1alpha1/providers/config/catalog/catalogprovider.go (hunk 5 above)
Comment (linyguo) [reply api/pkg/apis/v1alpha1/providers/config/catalog/catalogprovider.go]: add context in signature

File: api/pkg/apis/v1alpha1/utils/parser.go (hunk 6 above)
Comment (linyguo) [reply api/pkg/apis/v1alpha1/utils/parser.go:784]: yes it is from evaluation context.

File: api/pkg/apis/v1alpha1/utils/parser.go (hunk 7 above)
Comment (linyguo) [reply api/pkg/apis/v1alpha1/utils/parser.go:1460]: add logging in parser.Eval()

**Review by iwangjintian — APPROVED** [review]

File: api/pkg/apis/v1alpha1/managers/secrets/secrets-manager.go [hunk 8]
```diff
@@ -109,6 +110,8 @@ func (s *SecretsManager) Get(object string, field string, localContext interface
 			}
 		}
 	}
+
+	log.ErrorfCtx(ctx, " M (secret): No provider found for object: %s", object)
```
Comment (LeiQL) [inline api/pkg/apis/v1alpha1/managers/secrets/secrets-manager.go:114]: Keep upper case stype for `M(xxx)`

File: api/pkg/apis/v1alpha1/managers/jobs/jobs-manager.go [hunk 9]
```diff
@@ -524,7 +524,7 @@ func (s *JobsManager) HandleJobEvent(ctx context.Context, event v1alpha2.Event)
 				return err
 			}
 			var deployment model.DeploymentSpec
-			deployment, err = utils.CreateSymphonyDeploymentFromTarget(target, namespace)
+			deployment, err = utils.CreateSymphonyDeploymentFromTarget(ctx, target, namespace)
 			if err != nil {
```
Comment (LeiQL) [inline api/pkg/apis/v1alpha1/managers/jobs/jobs-manager.go:528]: Add logs for error situation as I see no log in `Vendor` for the errors thrown.

File: api/pkg/apis/v1alpha1/managers/secrets/secrets-manager.go (hunk 8 above)
Comment (linyguo) [reply api/pkg/apis/v1alpha1/managers/secrets/secrets-manager.go]: updated

File: api/pkg/apis/v1alpha1/managers/jobs/jobs-manager.go (hunk 9 above)
Comment (linyguo) [reply api/pkg/apis/v1alpha1/managers/jobs/jobs-manager.go:528]: updated

**Review by msftcoderdjw — APPROVED** [review]

**Pull request merged**